### PR TITLE
Add assert.retry function in common.js

### PR DESCRIPTION
**Pull request by MohitSuhane**

### CODE REVIEW

File: lib/common.js [hunk 1]
```diff
@@ -82,3 +82,26 @@ exports.assert.shouldThrowGen = function* (func, errorObject, message) {
   if (message === undefined) message = '';
   chai.assert(false, `${message}. Should have thrown error: ${errorObject.toString()} instead got ${JSON.stringify(result)}`);
 }
+
+/* execute a generator function that retry the input func until,
+ the func is executed with some return value or timeout with expected error */
+exports.assert.retry = function* (func, timeoutSeconds, expectedError) {
```
Comment (liorsaar1) [inline lib/common.js]: Retry a generator function (func) for a given timeout (timeoutSeconds). Allow an optional error (expectedError).

File: lib/common.js [hunk 2]
```diff
@@ -82,3 +82,26 @@ exports.assert.shouldThrowGen = function* (func, errorObject, message) {
...
+      }
+      // check expected error
+      const status = err.response.error.status;
+      chai.assert.equal(status, expectedError, 'Error must be ' + expectedError);
+    }
+    // sleep
+    yield util.sleep(1 * 1000);
+  }
+  chai.assert(false, 'timeout');
+}
```
Comment (liorsaar1) [inline lib/common.js]: chai.assert(false, 'assert.retry() timeout: ' + timeoutSeconds );

File: lib/common.js (hunk 1 above)
Comment (MohitSuhane) [reply lib/common.js]: Done

File: lib/common.js (hunk 2 above)
Comment (MohitSuhane) [reply lib/common.js]: Done

File: lib/common.js (hunk 1 above)
Comment (MohitSuhane) [reply lib/common.js]: Done

File: lib/common.js (hunk 2 above)
Comment (MohitSuhane) [reply lib/common.js]: Done

**Pull request merged**